atomkraft/test/test.py

```diff
@@ -163,7 +150,8 @@ def save_validator_files(self, val_root_dir: Path, report_dir: Path):
 
         for (trace_path, validator_dir) in zip(self.trace_paths, vals_dirs):
             copy_if_exists(
-                [trace_path, validator_dir], report_dir / Test._path_to_id(trace_path)
+                [trace_path, validator_dir],
+                report_dir / Path(Test._path_to_id(Path(os.path.basename(trace_path)))),
```

Comment:
maybe it doesn't make sense to pass `report_dir` separately here. Just use it as `self.report_dir`.

Reply:
btw, there should be a comment before the loop. This loop is not ideal and needs to be changed. It assumes we iterated over `self.trace_paths` in sequences, and the validator directories were last modified in that order. The comment should document it.

I can add that comment in this PR, if you don't mind.

Reply:
> This loop is not ideal and needs to be changed.

Right. I refactor this code from the previous implementation and I didn't notice it. I added a comment.